### ISSUE TITLE
>= 4.11.2 findOneAndUpdate behaves in differently with DateOnly package

### ISSUE THREAD

**Issue by misleadingTitle** [Automattic/mongoose#5536]
After version 4.11.1 the following code behaves differently:
```js
var mongoose = require("mongoose");
var DateOnly = require('mongoose-dateonly')(mongoose);

mongoose.connect('mongodb://localhost:27017/test');
var db = mongoose.connection;
db.on('error', console.error.bind(console, 'connection error:'));
db.once('open', function () {
    console.log("Connected...");
});

schema = mongoose.Schema({
    dateOfBirth: DateOnly
});

var testData = {
    "dateOfBirth": 20170809
};

var model = mongoose.model("test", schema);
model.findOneAndUpdate({
    "dateOfBirth": 20170809
}, testData, {upsert: true, new: true, passRawResult: true}, function (err, data, rawData) {
    var result = data.toObject();
    console.log(data.dateOfBirth);
});
```

Version 4.11.1 output:
```
[...]
Connected...
{ [Number: 20170809] date: 9, month: 8, year: 2017 }
```

Version 4.11.2 output:
```
[...]
Connected...
{ [Number: 20170709] date: 9, month: 7, year: 2017 }
```

Note that between versions the month numeration has changed from 8 (which is correct) to 7 (Javascript weird month indexing). The DateOnly package version has not been changed and the package is quite simple.

Why this is happening? I want to know in order to investigate if other sections of my project could be influenced by this behavior.

**Comment by varunjayaraman** [comment]
hmm, with your repro script, the log output for me is 

```javascript
Connected...
{ [Number: 20170717] date: 17, month: 7, year: 2017 }
```

That's for mongoose 4.11.2

For mongoose 4.11.1, I get what you got:

```javascript
Connected...
{ [Number: 20170809] date: 9, month: 8, year: 2017 }
```

I'm not too familiar with `dateonly` and it seems like `mongoose-dateonly` has a further dependency on `dateonly`. What version of `mongoose-dateonly` are you using?

**Comment by misleadingTitle** [comment]
I'm using version 1.0.3 of mongoose-dateonly. Both dateonly and mongoose-dateonly has not been updated in a long time.
I've tested to verify the odd incongruities between your results and mine and I've found that with mongoose >=4.11.2 the result is always set on the current day date, ignoring the value passed in dateOfBirth.

For a test I've set my system date and time in the future (August 24th 2018) and the result for mongoose 4.11.2 was
```
Connected...
{ [Number: 20180724] date: 24, month: 7, year: 2018 }
```
while on mongoose 4.11.1 the result was coherent with the one presented in my previous post.

In my tests I've only changed mongoose package version, while using the latest for mongoose-dateonly.
